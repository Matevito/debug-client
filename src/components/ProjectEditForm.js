import React, { useState, useEffect } from 'react'

// mui components
import {
    Button,
    TextField,
    FormGroup,
    FormControlLabel,
    Checkbox,
    Typography,
    Select,
    MenuItem
} from "@mui/material"

// app components
import { AuthError } from "./AuthError"

export const ProjectEditForm = ({usersList, handleSubmit, project, errors}) => {
    // form state values;
    const [title, setTitle] = useState("");
    const [description, setDescription] = useState("");
    const [team, setTeam] = useState([]);
    const [teamLeader, setTeamLeader] = useState("");
    
    // chekBox states
    const [teamState, setTeamState] = useState([]);

    useEffect(() => {
        if (project && usersList) {
            // set up project values;
            setTitle(project.data.title);
            setDescription(project.data.description);
            setTeamLeader(project.data.teamLeader._id)

            // set up team values;
            const teamIds = project.data.team.map((user) => user._id)
            setTeamState(
                usersList.map((user) => {
                    const userId = user.id;
                    console.log(userId)
                    if (teamIds.includes(userId)){
                        return true
                    } else {
                        return false
                    }
                })
            )
            setTeam(teamIds)
        }
        
    }, [project, usersList])

    const handleTeamBoxes = (position) => {
        // update chekbox value
        const updatedTeamState = teamState.map((item, index) => {
            return index === position ? !item : item
        });
        setTeamState(updatedTeamState);

        // change team state
        const selectedUser = {
            user: usersList[position],
            value: !teamState[position]
        };
        
        if (team.length === 0 && selectedUser.value === true) {
            // if there are no team users assigned yet.
            setTeam([selectedUser.user.id]);
        } else {
            if ( selectedUser.value === true) {
                // add users to team
                const newTeam = [ ...team, selectedUser.user.id];
                setTeam(newTeam)
            } else {
                // remove user from team
                const newTeam = team.filter((id) => {
                    if (id !== selectedUser.user.id) return id
                });
                setTeam(newTeam);
            }
        };

        // if there's no team, a teamLeader cannot be selected
        if (team.length === 0) {
            setTeamLeader("");
        }

        // if a team leader is removed from team, set teamLeader as null
        if (selectedUser.value === false && teamLeader === selectedUser.user.id){
            setTeamLeader("");
        }
    };
    const handleMultipleSelect = (e) => {

    };
    const handleForm = (e) => {
        e.preventDefault();
        const formObject = {
            title,
            description,
            team,
            teamLeader
        };
        handleSubmit(formObject);
    }
    return (
        <>
            <form action="#" onSubmit={handleForm}>
                <AuthError error={errors} />
                <TextField 
                    fullWidth
                    label="Title"
                    value={title}
                    onChange={(e) => { setTitle(e.target.value) }}
                    margin="dense"
                />
                <TextField 
                    fullWidth
                    label="Description"
                    value={description}
                    onChange={(e) => { setDescription(e.target.value)}}
                    variant="standard"
                    multiline
                    margin="dense"
                />

                <Typography>
                    Team members
                </Typography>
                <FormGroup
                    row={true}
                >
                    {usersList.map((user, index) => {
                        // check value to true if user is in the team
                        return (
                            <FormControlLabel
                                key={index}
                                label={user.username}
                                control={
                                    <Checkbox 
                                        value={user.id}
                                        checked={team.includes(user.id) ? true : false}
                                        onChange={() => {
                                            handleTeamBoxes(index)
                                        }}
                                    />
                                }
                            />
                        )
                    })}
                </FormGroup>
            </form>
        </>
    )
}
