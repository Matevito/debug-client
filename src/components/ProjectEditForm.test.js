import React from "react";

// testing utilites
import { fireEvent, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

// test comp
import { ProjectEditForm } from "./ProjectEditForm";

describe("ProjectEditForm component", () => {
    let projectData;
    let usersList;
    beforeEach(() => {
        projectData = {
            data:{
                _id: "projectID",
                title: "test title",
                description: "a description",
                team: [
                    {_id: "one", username:"user1"},
                    {_id:"two", username:"user2"}
                ],
                teamLeader: {
                    _id: "one", username:"user1"
                }
            },
            issues: []
        };
        usersList = [
            {id: "one", username:"user1"},
            {id:"two", username:"user2"},
            {id:"tres", username: "user3"}
        ];

    })
    test("renders correct form", () => {
        const { container } = render(<ProjectEditForm usersList={usersList} project={projectData}/>)
        expect(container).toMatchSnapshot();
    });
    test("handles submit", () => {
        const mockSubmit = jest.fn();
        render(
            <ProjectEditForm 
                usersList={usersList}
                project={projectData}
                handleSubmit={mockSubmit}
            />
        )

        const btn = screen.getByText("Edit project!");
        fireEvent.click(btn);
        const formRes = mockSubmit.mock.lastCall[0];

        expect(formRes.title).toBe(projectData.data.title);
        expect(formRes.description).toBe(projectData.data.description);
        expect(formRes.team).toEqual(["one", "two"]);
        expect(formRes.teamLeader).toBe(projectData.data.teamLeader._id);
    });
})