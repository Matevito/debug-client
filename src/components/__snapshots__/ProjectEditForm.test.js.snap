// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectEditForm component component handles submit 1`] = `
<div>
  <form
    action="#"
  >
    <div
      class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for=":r0:"
        id=":r0:-label"
      >
        Title
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r0:"
          type="text"
          value="test title"
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1in441m"
          >
            <span>
              Title
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for=":r1:"
        id=":r1:-label"
      >
        Description
      </label>
      <div
        class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline css-68u1dt-MuiInputBase-root-MuiInput-root"
      >
        <textarea
          aria-invalid="false"
          class="MuiInput-input MuiInputBase-input MuiInputBase-inputMultiline css-66dh3a-MuiInputBase-input-MuiInput-input"
          id=":r1:"
          style="height: 0px; overflow: hidden;"
        >
          a description
        </textarea>
        <textarea
          aria-hidden="true"
          class="MuiInput-input MuiInputBase-input MuiInputBase-inputMultiline css-66dh3a-MuiInputBase-input-MuiInput-input"
          readonly=""
          style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
          tabindex="-1"
        />
      </div>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Team members
    </p>
    <div
      class="MuiFormGroup-root MuiFormGroup-row css-qfz70r-MuiFormGroup-root"
    >
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate="false"
            type="checkbox"
            value="one"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          user1
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate="false"
            type="checkbox"
            value="two"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          user2
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate="false"
            type="checkbox"
            value="tres"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          user3
        </span>
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth  css-1ualgfl-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          role="button"
          tabindex="0"
        >
          user1
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          tabindex="-1"
          value="one"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-hdw1oc"
          >
            <span
              class="notranslate"
            >
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-r13yb-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        Edit project!
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </form>
</div>
`;

exports[`ProjectEditForm component renders correct form 1`] = `
<div>
  <form
    action="#"
  >
    <div
      class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for=":r0:"
        id=":r0:-label"
      >
        Title
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r0:"
          type="text"
          value="test title"
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1in441m"
          >
            <span>
              Title
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for=":r1:"
        id=":r1:-label"
      >
        Description
      </label>
      <div
        class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline css-68u1dt-MuiInputBase-root-MuiInput-root"
      >
        <textarea
          aria-invalid="false"
          class="MuiInput-input MuiInputBase-input MuiInputBase-inputMultiline css-66dh3a-MuiInputBase-input-MuiInput-input"
          id=":r1:"
          style="height: 0px; overflow: hidden;"
        >
          a description
        </textarea>
        <textarea
          aria-hidden="true"
          class="MuiInput-input MuiInputBase-input MuiInputBase-inputMultiline css-66dh3a-MuiInputBase-input-MuiInput-input"
          readonly=""
          style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
          tabindex="-1"
        />
      </div>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Team members
    </p>
    <div
      class="MuiFormGroup-root MuiFormGroup-row css-qfz70r-MuiFormGroup-root"
    >
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate="false"
            type="checkbox"
            value="one"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          user1
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate="false"
            type="checkbox"
            value="two"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          user2
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate="false"
            type="checkbox"
            value="tres"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          user3
        </span>
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth  css-1ualgfl-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          role="button"
          tabindex="0"
        >
          user1
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          tabindex="-1"
          value="one"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-hdw1oc"
          >
            <span
              class="notranslate"
            >
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-r13yb-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        Edit project!
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </form>
</div>
`;
